pkg/util/cgroup/manager/cgroup.go

```diff
@@ -292,7 +292,54 @@ func DropCacheWithTimeoutWithRelativePath(timeoutSecs int, absCgroupPath string,
 	_, err := exec.Command("bash", "-c", cmd).Output()
 
 	delta := time.Since(startTime).Seconds()
-	general.Infof("[DropCacheWithTimeoutWithRelativePath] it takes %v to do \"%s\" on cgroup: %s", delta, cmd, absCgroupPath)
+	general.Infof("[DropCacheWithTimeoutAndAbsCGPath] it takes %v to do \"%s\" on cgroup: %s", delta, cmd, absCgroupPath)
+
+	// if this command timeout, a none-nil error will be returned,
+	// but we should return error iff error returns without timeout
+	if err != nil && int(delta) < timeoutSecs {
+		return err
+	}
+
+	return nil
+}
+
+func SetExtraCGMemLimitWithTimeoutAndRelCGPath(ctx context.Context, relCgroupPath string, timeoutSecs int, nbytes int64) error {
+	memoryAbsCGPath := common.GetAbsCgroupPath(common.CgroupSubsysMemory, relCgroupPath)
+
+	err := SetExtraCGMemLimitWithTimeoutAndAbsCGPath(timeoutSecs, memoryAbsCGPath, nbytes)
+	_ = asyncworker.EmitAsyncedMetrics(ctx, metrics.ConvertMapToTags(map[string]string{
```

Comment:
should this metric logic be done in asyncworker's scope? what if this method wasn't called by asyncworker?

Reply:
[TODO]
1. use async worker framework to record succees information for works
2. pass and use normal emitter to record specic information for works